### PR TITLE
change(tinyusb): Added repo templates for issue and pull request

### PR DESCRIPTION
**Pull request by roma-jam**
**Describe the PR**
Added the templates to: 
- Bug Report

and Pull Requests: 

- Update Component

**Additional context**
To simplify the PR workflow and keep all PRs in the same format.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (tore-espressif) [inline .github/ISSUE_TEMPLATE/bug-report.yml]: Should this be in esp-usb repository?

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (peter-marcisovsky) [inline .github/ISSUE_TEMPLATE/bug-report.yml]: This link is for `esp-usb` github issues

File: .github/ISSUE_TEMPLATE/feature-request.yml
Comment (peter-marcisovsky) [inline .github/ISSUE_TEMPLATE/feature-request.yml]: Here as well, link to `esp-usb` github

**Comment by roma-jam** [comment]
@tore-espressif @peter-marcisovsky 
Thanks for such a quick review, but I basically just copied files from the esp-usb, because I wanted to check it.
Now, while I still have concerns, I will request for the review when it will be ready.

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (roma-jam) [reply .github/ISSUE_TEMPLATE/bug-report.yml]: espressif/tinyusb repo doesn't have the "Issues" itself. 

I asked to add them and changed the link accordingly. 
Thanks.

File: .github/ISSUE_TEMPLATE/feature-request.yml
Comment (roma-jam) [reply .github/ISSUE_TEMPLATE/feature-request.yml]: I removed this template as I don't see it can be helpful.
Just based on the rate how often we get such feature-requests in espressif/tinyusb.

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (roma-jam) [reply .github/ISSUE_TEMPLATE/bug-report.yml]: I removed that part. 
Thanks for checking it.

**Comment by roma-jam** [comment]
@tore-espressif @peter-marcisovsky 
I have resolved the problem with Pull Request template (nested templates are not working ATM) and removed the issues templates, which I think we do not need. 

PTAL again.

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (peter-marcisovsky) [inline .github/ISSUE_TEMPLATE/bug_report.yml]: I suppose, the `Issues` tab will be created in this repo after merging this changes, as right now the tab is missing and this link gets you to `Pull requests` tab.

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (peter-marcisovsky) [inline .github/ISSUE_TEMPLATE/bug_report.yml]: I think, here should be either 
```suggestion
      description: Which Development Kit does this issue occur on?
```

or

```suggestion
      description: On which Development Kit does this issue occur?
```

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (peter-marcisovsky) [inline .github/ISSUE_TEMPLATE/bug_report.yml]: Likewise here.. slight redundancy of 'on'

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (peter-marcisovsky) [inline .github/ISSUE_TEMPLATE/bug_report.yml]: Likewise here.. slight redundancy of 'on'

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (peter-marcisovsky) [inline .github/ISSUE_TEMPLATE/bug_report.yml]: Optionally: we can add a word issue, or a bug.. or something similar. Just to be sure  
```suggestion
      description: Do you have any other information from investigating this issue?
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (peter-marcisovsky) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: Shouldn't the `PULL_REQUEST_TEMPLATE` be a folder name, inside of which an whatever `*.md` file resides? Just looking at the esp-usb repo, where we have it like this.

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (roma-jam) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: No, Issues tab should be created by maintainers. 

So, the questions is: Do we want split the issues for espressif/tinyusb? 

Basically, right now, there are three options: 
1. Use `esp-usb` issue tracker
2. Use `esp-idf` issue tracker
3. Create and use `espressif/tinyusb` issue tracker

I guess that this questions is being asked all the time, when new commiter comes.
Anyway, @tore-espressif do we have any special requirements why we didn't create the issue tracker for `espressif/tinyusb`?
Also, please, share your opinion about question which tracker to use. Thanks.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (roma-jam) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: ATM, the nesting templates for pull requests are not working. 
At least that what I was able to find and after removing the folder everything started to work well. 

Also, there was a `pull_request_template.md` in the `.github` folder, but I decided to keep it with uppercase names, as when there are lots of files it can be easily missed.

Comment (peter-marcisovsky) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: Do you think, it might be the same issue with the magic PR template we are seeing in esp-usb? So we will just have to rename/restructure the `.md` files, for our PR template to appear?

Comment (roma-jam) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: @peter-marcisovsky IDK, I haven't check it in esp-usb, but for other my personal repo it worked.

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (peter-marcisovsky) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Oh sry, I haven't thought about the issues tracker in the tinyusb upstream. 

I took a look at the other espressif github repos, which are forked, just like espressif/tinyusb, [here](https://github.com/espressif?q=&type=fork&language=&sort=). 

Some espressif forks have the issue trackers, some don't. Sometimes, even the upstream repo does not have the issue tracker, so the espressif fork has to have one. Looks like, it's rather a personal choice to have one.

**Comment by roma-jam** [comment]
@tore-espressif @peter-marcisovsky 
Thanks for the reviewing. 
I have updated the links to issues trackers (re-direct the user to upstream issues or to esp-usb issue tracker) and also added the `issues`  optional field to `idf_component.yml`

PTAL.

**Review by peter-marcisovsky — APPROVED** [review]
Thx for the changes. LGTM

**Review by tore-espressif — APPROVED** [review]

**Pull request merged**
